### PR TITLE
feat: update to `jiti` v2

### PR DESCRIPTION
**Pull request by aryaemami59**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain:

`jiti` v2 has been [released](https://github.com/unjs/jiti/releases/tag/v2.0.0), which means we can now update the syntax, documentation and unit tests to be compatible with its latest release.

#### What changes did you make? (Give an overview)

**This PR**:

- [X] Bumps `jiti` to version 2.
- [X] Uncomments unit tests related to `jiti` v2 and top-level `await` syntax.
- [X] Removes warning about using top-level `await` syntax in TypeScript config files.
- [X] Uses `jiti.createJiti` instead of the deprecated `default` export.

### CODE REVIEW

**Review by nzakas — APPROVED** [review]
LGTM. Would like another review before merging.

**Review by aladdin-add — APPROVED** [review]
LGTM. Waiting a few days before merging.

File: lib/config/config-loader.js [hunk 1]
```diff
@@ -173,7 +173,7 @@ async function loadConfigFile(filePath, allowTS) {
      */
     if (allowTS && isTS && !isDeno && !isBun) {
 
-        const createJiti = await import("jiti").then(jitiModule => jitiModule.default, () => {
+        const createJiti = await import("jiti").then(jitiModule => (typeof jitiModule?.createJiti === "function" ? jitiModule.createJiti : jitiModule.default), () => {
```
Comment (fasttime) [inline lib/config/config-loader.js:176]: This will not throw an error if jiti v1.26 is installed, correct?

Comment (aryaemami59) [reply lib/config/config-loader.js:176]: Nope, just checked it's fine.

**Review by fasttime — APPROVED** [review]
LGTM, thanks!

**Pull request merged**